### ISSUE TITLE
[BUG] Unable to request token with custom scope

### ISSUE THREAD

**Issue by ganniaiuei** [Azure/azure-sdk-for-net#34935]
### Library name and version

Azure.Identity 1.8.2, Microsoft.NET.Sdk.Functions 3.1.2

### Describe the bug

I'm using the Azure.Identity from a Function App (runtime 3) calling the DefaultAzureCredential() class. I've already activated the Managed Identity for the Function App. I'm calling the GetTokenAsync to gather a token for another app which exposes an API using this code. The scope is in the format "api://someguid/.default"
```
var scope = "api://<guid>/.default"   
var tokenCredential = new DefaultAzureCredential();   
var tokenResponse = await tokenCredential.GetTokenAsync(   
  new TokenRequestContext(new[] { scope }),   
  cancellationToken   
);  
```
The API I'm trying to get access to, is authenticated using the Easy Auth Authentication from the portal. This auth results in a new app registration created in the Azure Active Directory. I've also created App Roles for the API and assigned an app role to my Managed Identity using the powershell command `New-AzureADServiceAppRoleAssignment`

### Expected behavior

I would expect to retrieve a token, or at least have a more specific error than the 500. 

### Actual behavior

The code throws an exception when calling the GetTokenAsync with a 500 Internal Server error. Azure.Identity.AuthenticationFailedException : ManagedIdentityCredential authentication failed: Service request failed.Status: 500 (Internal Server Error).
As a side note, if I change the scope to e.g. "https://<your-key-vault-name>.vault.azure.net/" it gets the token properly.

### Reproduction Steps

Create an app service with Easy Auth activated - name: MyAPIApp
Get the App Registration linked to the Easy Auth of the MyAPIApp
Create a new Azure Function with HttpTrigger in Visual Studio.
Import the Assembly Azure.Identity
Retrieve the token for the ManagedIdentity using the code above and the scope from the App Registration of the MyAPIApp
Publish the Function App to Azure - Name : MyFunctionApp
Run the Trigger Http to execute the Function
The error 500 appears

### Environment

Microsoft Visual Studio Enterprise 2019 v. 16.10.3
Azure Function runtime 3.x

**Comment by jsquire** [comment]
//cc: @christothes

**Comment by christothes** [comment]
Hi @ganniaiuei  - Does this work if you specify the ClientId of your application in the `DefaultAzureCredentialOptions` here:?

https://github.com/Azure/azure-sdk-for-net/blob/1f2380dc1864f41809530e8366c2f97f8ffd9e76/sdk/identity/Azure.Identity/src/Credentials/DefaultAzureCredentialOptions.cs#L196

**Comment by christothes** [comment]
Just to be clear - is `my-managed-id` the "Application (client) ID" of the application you are authenticating to?

**Comment by ganniaiuei** [comment]
You're right, I got the wrong id (the objectID), now I tried with the ApplicationID (EnterpriseApp->ManagedIdentities->ApplicationID). But the error is still the same: 400 Bad Request

**Comment by christothes** [comment]
OK - Sorry, I was confused about the clientId option. That ManagedIdentityClientId is meant for addressing a user defined managed identity. So as long as your managed identity is system assigned, you don't need to specify it. The scope alone should be enough to address your application.

After removing the clientID from the options, does the content logging provide any additional details about the 500 error coming back?

**Comment by ganniaiuei** [comment]
After removing the ManagedClientId I run the app locally with verbose logging and I got the message that the app with api://guid/.user_impersonation wasn't found in my tenant. So I just removed the /.user_impersonation and things started to change. I noticed that the proper scope was: api://guid/.default or only api://guid. After that, I added the ManagedIdentity to the users allowed to run the Enterprise App (the APIApp have the Assignment Required to yes) and the token was properly issued. Thank you for letting me dig in to the issue. I'll raise an issue to the docs as the specific flow (App Identity and Permissions) may be described a little be better (IMHO)
